File: ServidorCliente/src/Hilo.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.net.Socket;
import java.sql.ClientInfoStatus;

public class Hilo extends Thread{

    private Socket cliente = null;

    public Hilo(Socket cliente) {
        this.cliente = cliente;
    }

    @Override
    public void run() {
        super.run();
        try {
            PrintWriter out = new PrintWriter(cliente.getOutputStream(),
                    true);
            BufferedReader in = new BufferedReader
                    (new InputStreamReader(cliente.getInputStream()));
            String inputLine = in.readLine();

            System.out.println("Recibido: " +inputLine);
            out.println("Nos ha llegado el mensaje");

            out.close();
            in.close();
            cliente.close();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: EntregaTres/Cliente/app/src/main/java/com/example/cliente/Hilo.kt
package com.example.cliente

import java.io.PrintWriter
import java.net.Socket


class Hilo(var ip:String, var puerto:Int,var mensaje:String) : Thread() {
    //PARA QUE FUNCIONE Y NO SALTE "socket failed: eperm (operation not permitted)"
    // AÑADIR AL AndroidManifest.xml la siguiente linea
        //<uses-permission android:name="android.permission.INTERNET"/>

    private lateinit var client: Socket
    private lateinit var out: PrintWriter

    override fun run() {
        super.run()
        client = Socket(ip, puerto)
        out = PrintWriter(client.getOutputStream(), true)
        out.println(mensaje)
        out.close()
        client.close()
        //"192.168.1.128", 1234
    }


}